src/prefect/tasks.py

```diff
@@ -77,6 +77,25 @@ def task_input_hash(
     )
 
 
+def exponential_backoff(backoff_factor: float) -> Callable[[int], List[float]]:
+    """
+    A task retry backoff utility that configures exponential backoff for task retries.
+    The exponential backoff design matches the urllib3 implementation.
+
+    Arguments:
+        backoff_factor: the base delay for the first retry, subsequent retries will
+            increase the delay time by powers of 2.
+
+    Returns:
+        a callable that can be passed to the task constructor
+    """
+
+    def retry_backoff_callable(retries: int) -> List[float]:
+        return [backoff_factor * max(0, 2**r) for r in range(retries)]
```

Comment:
Let's limit this to our maximum length so our own utility doesn't go out of bounds.